### DISCUSSION TITLE
Node.js and npm update

### DISCUSSION THREAD

**Discussion by Quitomm** [coder/code-server#6949] [Q&A]
I installed the code-server manually by downloading the build from the release tab. While working, I found out that the code-server uses nodejs v12. How can I update/rebuild?

**Comment by code-asher** [comment]
v12?  It should be bundling v20, if you are installing the latest.

If you want to use a different version of Node you will have to recompile code-server.  The easiest way to do that is to install from npm instead.  You can set `FORCE_NODE_VERSION` to use a different version.

This is not recommended though, changing the Node version can break code-server.

**Reply by Quitomm** [reply]
Yes, I have installed the latest version of the code server. I downloaded the archive and unpacked it on my server. When checking node -v via code-server, it outputs v12(I also had v12 nodejs installed on my server). I have updated nodejs on my server to v20, but when checking through code-server it still outputs v12

**Reply by code-asher** [reply]
Oh your system might be using v12, but code-server ships with its own Node binary that it will use (found in `lib/node`).

We do not set `PATH` or anything, so inside the terminal it will still show your system's Node, but code-server itself should be using v20.

For example, on my system, where I have code-server running from `/tmp/code-server`:

```
$ ps -eFww | grep code-server
coder       1445       1  0 275110 63744 19 Aug21 ?        00:00:00 /tmp/code-server/lib/code-server-4.92.2/lib/node /tmp/code-server/lib/code-server-4.92.2  --auth none --port 13337 --app-name code-server
coder       1468    1445  0 329992 90984 26 Aug21 ?        00:00:05 /tmp/code-server/lib/code-server-4.92.2/lib/node /tmp/code-server/lib/code-server-4.92.2/out/node/entry
coder      73957   73708  0   987  2340  43 00:59 pts/0    00:00:00 grep --color=auto code-server
$ /tmp/code-server/lib/code-server-4.92.2/lib/node --version
v20.11.1
```
